include/seastar/core/abort_source.hh

```diff
@@ -22,20 +22,25 @@
 #pragma once
 
 #include <seastar/util/concepts.hh>
+#include <seastar/util/modules.hh>
 #include <seastar/util/noncopyable_function.hh>
 #include <seastar/util/optimized_optional.hh>
 #include <seastar/util/std-compat.hh>
-#include <boost/intrusive/list.hpp>
 
+#ifndef SEASTAR_MODULE
+#include <boost/intrusive/list.hpp>
 #include <exception>
 #include <optional>
 #include <type_traits>
 #include <utility>
+#endif
```

Comment:
I don't understand this #ifndef. How does it compile? and why shouldn't we include the headers, as long as we don't export the contents?

Reply:
explained in the commit message. i am quoting it here

> in the case of source files, because they are not included by seastar.cc, and they are compiled individually, we need to keep the used third-party headers in its global fragment. but please note, the required headers might be more than what we generally need when compiling the source file to a regular library. because, unlike the textual inclusion, the source file will not include the headers previously included by the Seastar headers anymore. now, the .cc files only have access to the symbols attached "seastar" module by default. so we need to include them explicity in the global fragment. that's why a bunch of "new" includes are added in this change.